text/2020-04-23-rawkv-txnkv-mode-switching.md

```diff
@@ -0,0 +1,63 @@
+# Title
+
+## Summary
+
+This RFC proposes to forbid users using RawKV API on the TiKV cluster by default,
+and provide a new API command to let users enable it.
+
+## Motivation
+
+Some painful incidents happened before and still happening,
+because of the mixed-using of RawKV API and TransactionKV API,
+some key-values were overwritten by RawKV API,
+which originally written by TransactionKV API,
+then led to data corruption.
+
+We should add more warnings about this in our documents, but it's not enough,
+documents could easily be ignored.
+
+Providing a switching API could prevent people from misuse without noticing,
+when users need to step into the danger zone,
+they need to explicitly open the guarded switch.
+And they could shut down the switch anytime to gain more safety.
+
+## Detailed design
+
+This `mode` flag should be persisted,
+each member of the TiKV cluster should see the same value in the same time,
+so it should be write/read by RawKV commands.
+Or we just use the dynamic configuration feature to achieve our goal.
+
+`mode` should be an enum:
+
+- `ModeRawKV` indicate that the TxnKV commands could not be called.
+  When they are called they will return errors.
+- `ModeTxnKV` indicate that the RawKV commands could not be called.
+- `ModeTxnKVRawKV` works as now.
```

Comment:
This will mislead users to mix using rawkv and txnkv, I suggest to remove it.

Reply:
Don't we need this for backwards compatabilty?

Reply:
@nrc yes it just means for backward compatibility, if @zhangjinpeng1987 can confirm it's not necessary, then we should remove it